src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/visits/VisitSynchronisationService.kt

```diff
@@ -24,29 +25,36 @@ class VisitSynchronisationService(
 
       log.debug("received nomis visit: ${this@run}")
 
-      mappingService.findNomisVisitMapping(
-        visitCancelledEvent.visitId
-      )?.let { visitMapping ->
-        log.debug("found nomis visit mapping: $visitMapping")
-        val vsipOutcome = getVsipOutcome(this@run) ?: VsipOutcome.CANCELLATION
-        visitService.cancelVisit(
-          visitReference = visitMapping.vsipId,
-          outcome = VsipOutcomeDto(vsipOutcome, "Cancelled by NOMIS")
-        )
-
-        telemetryClient.trackEvent(
-          "visit-cancellation-synchronisation",
-          mapOf(
-            "offenderNo" to this.offenderNo,
-            "vsipId" to visitMapping.vsipId,
-            "nomisVisitId" to visitMapping.nomisId.toString(),
-            "vsipOutcome" to vsipOutcome.name
-          ),
-          null
-        )
-      } ?: let {
-        log.debug("Ignoring visit cancellation event for ${this@run} as no nomis visit mapping found")
+      if (nomisCancellation()) {
+        mappingService.findNomisVisitMapping(
+          visitCancelledEvent.visitId
+        )?.let { visitMapping ->
+          log.debug("found nomis visit mapping: $visitMapping")
+          val vsipOutcome = getVsipOutcome(this@run) ?: VsipOutcome.CANCELLATION
+          visitService.cancelVisit(
+            visitReference = visitMapping.vsipId,
+            outcome = VsipOutcomeDto(vsipOutcome, "Cancelled by NOMIS")
+          )
+
+          telemetryClient.trackEvent(
+            "visit-cancellation-synchronisation",
+            mapOf(
+              "offenderNo" to this.offenderNo,
+              "vsipId" to visitMapping.vsipId,
+              "nomisVisitId" to visitMapping.nomisId.toString(),
+              "vsipOutcome" to vsipOutcome.name
+            ),
+            null
+          )
+        } ?: let {
+          log.debug("Ignoring visit cancellation event for ${this@run} as no nomis visit mapping found")
+        }
+      } else {
+        log.debug("Ignoring VSIP-originated visit cancellation event for ${this@run}")
       }
     }
   }
+
+  private fun NomisVisit.nomisCancellation() =
+    this.modifyUserId != "PRISONER_MANAGER_API" // doesn't originate from VSIP
```

Comment:
I wonder if we should make this an environment property or anything? I think for now hardcode it but might be worth thinking if there is a better way